### DISCUSSION TITLE
Monitor GPU usage

### DISCUSSION THREAD

**Discussion by Mailiax** [ClementTsang/bottom#1067] [Q&A] [answered]
Can you monitor GPU usage and GPU memory usage in any way? If not, is it a limitation in sysinfo or something else?

**Comment by ClementTsang** [comment, marked as answer]
bottom currently supports GPU memory usage for Nvidia GPUs only, though it's behind a runtime flag/option for at least now (`--enable_gpu_memory`). A GPU usage widget would be nice as well and is a filed issue (#298) that I would love to do when I have more time (or would happily accept PRs for).

As for why not, say, AMD or Intel (yet), it's just down to getting the data. Sysinfo and heim provide nothing related to this. Nvidia GPUs can leverage nvml, for which [there are easy Rust bindings for](https://docs.rs/nvml-wrapper/latest/nvml_wrapper/?search=NVML). As for AMD or Intel, I don't know if there are any bindings I can use? If there aren't, I'll have to handle it myself, which is a bit of effort, especially to make things work for all supported platforms.

**Reply by ClementTsang** [reply]
And just to add, if anyone wants to add support, I'll gladly accept PRs.
